### PR TITLE
Update EIP-1559 label to "Fee optimization" for find-wallet

### PR DESCRIPTION
**Pull request by corwintines**
## Description
- Update label and description for `EIP-1559 support` to `Fee optimization`

### CODE REVIEW

File: src/data/wallets/wallet-filters.ts [hunk 1]
```diff
@@ -131,8 +131,9 @@ const walletFilterData: { [key: string]: WalletFilter } = {
     filterKey: "erc_20_support",
   },
   eip_1559_support: {
-    title: "EIP-1559 support",
-    description: "Lower gas fees with type-2 transactions",
+    title: "Fee optimization",
+    description:
+      "Support type 2 transactions for optimized gas fees, and fee refunds for unused gas",
```
Comment (samajammin) [inline src/data/wallets/wallet-filters.ts:136]: Questions (that could be solved in a separate PR):
- Would it still be illuminating/valuable to mention EIP 1559 anywhere?
- Do we define "type 2 transactions" anywhere? If not we probably should... e.g. on the tx page: https://ethereum.org/en/developers/docs/transactions/#typed-transaction-envelope

File: src/data/wallets/wallet-filters.ts [hunk 2]
```diff
@@ -131,8 +131,9 @@ const walletFilterData: { [key: string]: WalletFilter } = {
     filterKey: "erc_20_support",
   },
   eip_1559_support: {
```
Comment (samajammin) [inline src/data/wallets/wallet-filters.ts:133]: If we remove mention of EIP 1559, should we update this `eip_1559_support` key, e.g. to `type_2_tx_support`?

File: src/data/wallets/wallet-filters.ts (hunk 1 above)
Comment (samajammin) [reply src/data/wallets/wallet-filters.ts:136]: Created an issue around the 2nd point: https://github.com/ethereum/ethereum-org-website/issues/7372

Comment (corwintines) [reply src/data/wallets/wallet-filters.ts:136]: I think we could add something to the description for this. I was going to do a general pass over the copy for these next sprint so I think it could be done then.

File: src/data/wallets/wallet-filters.ts (hunk 2 above)
Comment (corwintines) [reply src/data/wallets/wallet-filters.ts:133]: Been looking at this code again and I think there should be a refactor on the filters, I think for myself keeping it as 
`eip_1559_support` helps as it is clear what it is as a variable, but open to suggestions

**Review by samajammin — APPROVED** [review]
